lib/orbf/rules_engine/services/ethiopian_period_iterator.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Orbf
+    module RulesEngine
+      class EthiopianPeriodIterator
```

Comment:
I think this might need a small comment explaining why the quarters are behaving the way they are and that it's normal that a Q1 will take the months of a previous year etc.